pkg/kubectl/cmd/clusterinfo.go

```diff
@@ -93,10 +94,24 @@ func RunClusterInfo(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error
 					link += "http://" + ip + ":" + strconv.Itoa(int(port.Port)) + " "
 				}
 			} else {
+				name := service.ObjectMeta.Name
+
+				if len(service.Spec.Ports) > 0 {
+					port := service.Spec.Ports[0]
+
+					// guess if the scheme is https
+					scheme := ""
+					if port.Name == "https" || port.Port == 443 {
+						scheme = "https"
+					}
+
+					name = utilnet.JoinSchemeNamePort(scheme, service.ObjectMeta.Name, port.Name)
```

Comment:
should the last parameter be `port.Port`?

Reply:
no, if you specify a number, that just gets turned back into the port name server-side:

https://github.com/kubernetes/kubernetes/blob/2986b37de06fcd1bf94e68b49b6c8d87c389d3c2/pkg/registry/core/service/rest.go#L388-L406

Reply:
Worth adding a comment? I thought this looked wrong too.

Reply:
Aha. thanks @liggitt